cranelift/codegen/src/machinst/abi.rs

```diff
@@ -106,7 +99,7 @@ pub trait ABICallee {
     fn set_num_spillslots(&mut self, slots: usize);
 
     /// Update with the clobbered registers, post-regalloc.
-    fn set_clobbered(&mut self, clobbered: Set<Writable<RealReg>>);
+    fn set_clobbered(&mut self, clobbered: Vec<Writable<RealReg>>);
```

Comment:
Sort of strange API-wise to give a `Vec` here -- is there a reason we aren't giving a `&[Writable<RealReg>]`? Do impls take ownership/mutate/grow the vec internally?

Reply:
The reason is to avoid an allocation: when we compute it, we incrementally build it in a `Vec`; the only use is to call `set_clobbered`, and the ABI object needs to subsequently retain it. So if we passed a slice we'd force a copy. Slightly non-idiomatic for the general API case but here the method just ties together two specific points in the code :-)